docs/root/intro/arch_overview/upstream/dns_resolution.rst

```diff
@@ -13,6 +13,11 @@ Envoy uses `c-ares <https://github.com/c-ares/c-ares>`_ as a third party DNS res
 On Apple OSes Envoy additionally offers resolution using Apple specific APIs via the
 ``envoy.restart_features.use_apple_api_for_dns_lookups`` runtime feature.
 
+Envoy provides DNS resolution through extensions. To use a specific DNS resolver, corresponding DNS library need to be enabled in
+:repo:source/extensions/extensions_build_config.bzl file. Appropriate Envoy static or dynamic configuration of
+the DNS extension points is also required. For example:
+:ref:'typed_dns_resolver_config <envoy_v3_api_field_config.cluster.v3.Cluster.typed_dns_resolver_config>'
```

Comment:
Sorry I should have commented on this before but can we refactor this documentation slightly. I would do the following:

```
Envoy provides DNS resolution through extensions, and contains 2 built-in extensions:
1) c-ares
2) Apple (iOS/macOS only)
```
And then go on to actually provide worked examples of what the typed config would look like? Then in the various API docs where you try to give an example, I would just say "for an example of a built-in typed configuration see here." and actually link to this portion of the docs with worked examples for both macOS/c-ares?

The reason I'm bring this up is I think this new config is going to trip up a lot of existing users and I want to make sure they have a worked example to help them.

Reply:
Based on the comments, I updated dns_resolution.rst and docs/root/configuration/http/http_filters/dynamic_forward_proxy_filter.rst. PTAL.